pkg/resource/deploy/deployment_executor.go

```diff
@@ -569,8 +564,7 @@ func (ex *deploymentExecutor) refresh(callerCtx context.Context, opts Options, p
 	// Refreshes have their own version of ignoring errors that clashes with ContinueOnError.
 	// Set ContinueOnError to false to avoid that clash.  We'll still get the correct error
 	// ignoring behaviour.
-	opts.ContinueOnError = false
-	stepExec := newStepExecutor(ctx, cancel, ex.deployment, opts, preview, true)
+	stepExec := newStepExecutor(ctx, cancel, ex.deployment, true)
```

Comment:
We can avoid this mutation by rewiring the logic in the code that checks these two flags.